compiler/pipes/collect-main-edges.cpp

```diff
@@ -621,8 +621,8 @@ void CollectMainEdgesPass::on_finish() {
   if (current_function->is_extern()) {
     return;
   }
-  if (!have_returns) {
-    // hack to work well with functions which always throws
+  if (!have_returns && !current_function->return_typehint) {
```

Comment:
I think this change was included in the other PR too.

Reply:
Yes, you're right. That PR would be merged first, and after rebasing, there will be no change here.
This change is necessary for some tests to pass.